huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go

```diff
@@ -73,12 +76,12 @@ func dataSourceIecFlavors() *schema.Resource {
 	}
 }
 
-func dataSourceIecFlavorsV1Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
+func dataSourceIecFlavorsRead(d *schema.ResourceData, meta interface{}) error {
+	cfg := meta.(*config.Config)
 
-	iecClient, err := config.IECV1Client(GetRegion(d, config))
+	iecClient, err := cfg.IECV1Client(common.GetRegion(d, cfg))
```

Comment:
```suggestion
	iecClient, err := cfg.IECV1Client(cfg.GetRegion(d))
```
Please check the other parts